src/packages/protoc-gen-openapiv2/update-hash.py

```
@@ -0,0 +1,166 @@
+        with urllib.request.urlopen(url) as response:
+            data = json.loads(response.read().decode())
+            return data["tag_name"]
+    except Exception as e:
+        print(f"Error fetching latest release: {e}")
+        sys.exit(1)
+
+def get_current_version():
+    """Extract current version from default.nix."""
+    nix_file = Path(__file__).parent / NIX_FILE
+    
+    with open(nix_file, 'r') as f:
```

Pylint (3.3.7)

[warning] Using open without explicitly specifying an encoding

(W1514)